### PR TITLE
fix(components): [input] formatter trigger onmount

### LINKED ISSUE #8721: [Component] [input] [input] formatter、parser 默认值不会被format，且输入值parser不生效得到的值是format之后的
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.25`
- Element Plus Version: `2.2.8`
- Browser / OS: `Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.0.0 Safari/537.36`
- Build Tool: `Vite`

### Related Component

- `el-input`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbmltcG9ydCB7IHJlZiwgdmVyc2lvbiBhcyB2dWVWZXJzaW9uIH0gZnJvbSAndnVlJ1xuaW1wb3J0IHsgdmVyc2lvbiBhcyBFcFZlcnNpb24gfSBmcm9tICdlbGVtZW50LXBsdXMnXG5pbXBvcnQgeyBFbGVtZW50UGx1cyB9IGZyb20gJ0BlbGVtZW50LXBsdXMvaWNvbnMtdnVlJ1xuXG5jb25zdCBtc2cgPSByZWYoJzEyMzQ1Njc4JylcblxuY29uc3QgZm9ybWF0dGVyID0gdmFsID0+IFN0cmluZyh2YWwpLnJlcGxhY2UoIC8oXFxkKSg/PShcXGR7M30pKyQpL2csICckMSwnIClcbmNvbnN0IHBhcnNlciA9IHZhbCA9PiB2YWwucmVwbGFjZSgvLC9nLCAnJylcbjwvc2NyaXB0PlxuXG48dGVtcGxhdGU+XG4gIDxoMT57eyBtc2cgfX08L2gxPlxuICA8ZWwtaW5wdXQgdi1tb2RlbD1cIm1zZ1wiIDpmb3JtYXR0ZXI9XCJmb3JtYXR0ZXJcIiA6cGFyc2VyPVwicGFyc2VyXCIgLz5cbiAgPHA+XG4gICAgPGVsLWljb24gY29sb3I9XCJ2YXIoLS1lbC1jb2xvci1wcmltYXJ5KVwiPjxFbGVtZW50UGx1cyAvPjwvZWwtaWNvbj5cbiAgICBFbGVtZW50IFBsdXMge3sgRXBWZXJzaW9uIH19ICsgVnVlIHt7IHZ1ZVZlcnNpb24gfX1cbiAgPC9wPlxuPC90ZW1wbGF0ZT5cbiIsImltcG9ydF9tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHt9XG59IiwiX28iOnt9fQ==)

### Steps to reproduce

输入默认值： 123456789
输入数据： 123456789

### What is Expected?

默认显示：123,456,789
输入得到值：123456789

### What is actually happening?

默认显示：123456789
输入得到值：123,456,789

### Additional comments

_(empty)_

### PR DESCRIPTION
**Pull request by faga295**
formatter trigger onmount

closed #8721 #7394

Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Comment by HeftyKoo** [comment]
20221215115042_rec_
It's not work after clear the initial value.

**Comment by faga295** [comment]
now it works

**Comment by faga295** [comment]
Someone can review this pr?

**Comment by HeftyKoo** [comment]
> now it works

> Someone can review this pr?

Sorry for replying late.  But these are some issues I mentioned above you should resolve first.

https://github.com/element-plus/element-plus/pull/10338#issuecomment-1352522971

Each time input may cause the `formatter` function to invoke twice.  Because user input may change the `nativeInputValue`， then the watcher would trigger the `setNativeInputValue` function.

**Comment by faga295** [comment]
it will not invoke setNativeInputValue twice

File: packages/components/input/src/input.vue [hunk 1]
```diff
@@ -337,17 +337,20 @@ const resizeTextarea = () => {
...
 const handleInput = async (event: Event) => {
   recordCursor()
 
-  let { value } = event.target as TargetElement
-
+  const { value } = event.target as TargetElement
+  let parsedValue
   if (props.formatter) {
-    value = props.parser ? props.parser(value) : value
-    value = props.formatter(value)
```
Comment (MrWeilian) [inline packages/components/input/src/input.vue:358]: hello, I think just delete this line of code that is ok. Is any other reason for adding the new variable `parsedValue`?

**Comment by HeftyKoo** [comment]
cc  @chenxch

**Comment by LuckyArdhika** [comment]
this my problem now

**Comment by tolking** [comment]
Thanks for your contribution.

This issue appears to have been fixed

**Pull request closed without merging**